### PR TITLE
Improve pull for transit/bootstrap

### PR DESCRIPTION
**Pull request by jhawk28**
Allow concurrent downloads (default to 2)
Verify existing files to the MD5 hash (skip if match)

### CODE REVIEW

**Comment by vishalchangrani** [comment]
Some context for all the reviewers - After the testnet network upgrade yesterday, access node operators who attempted to download the root checkpoint files experienced that the download was very slow.
Perhaps the root.checkpoint files have grown in size from the last upgrade. Also, the transit script downloads the sequentially.
@jhawk28 has this PR to make the transit script download the files in parallel expediting the process.
This will be very helpful for the upcoming mainnet network upgrade since execution node partners will also benefit from it.

File: cmd/bootstrap/cmd/pull.go [hunk 1]
```diff
@@ -62,15 +70,49 @@ func pull(cmd *cobra.Command, args []string) {
...
+				}
 			}
-		}
+		}(file)
 	}
+
+	wait.Wait()
+}
+
+func CalcMd5(outpath string) []byte {
```
Comment (jordanschalm) [inline cmd/bootstrap/cmd/pull.go:105]: It would be great to pull this out into `cmd/bootstrap/utils` so it isn't duplicated in `transit`

**Review by jordanschalm — APPROVED** [review]
New logic looks good - just the one suggestion to pull out `CalcMd5` into a utils package to avoid duplication.

File: cmd/bootstrap/cmd/pull.go (hunk 1 above)
Comment (jhawk28) [reply cmd/bootstrap/cmd/pull.go]: moved CalcMd5 to cmd/bootstrap/utils

**Comment by jhawk28** [comment]
I was able to run the tool against one of our testnet nodes. I found that the MD5 was not on some of the large files such as "root.checkpoint.014". Upping the concurrency to 10, it was able to download most of the files in 5 minutes.

I added a check so that it doesn't skip files that don't have an MD5 to be safe.

Sample command that I ran that was successful: `transit pull -b ./bootstrap/ -t testnet-51-execution -r access --concurrency 10 --timeout 15m`

**Comment by vishalchangrani** [comment]
Linter is flagging md5 as `G501: Blocklisted import crypto/md5: weak cryptographic primitive (gosec)`. Is Sha256 better?

**Comment by jhawk28** [comment]
> Linter is flagging md5 as `G501: Blocklisted import crypto/md5: weak cryptographic primitive (gosec)`. Is Sha256 better?

The google storage API only provides md5 and crc32: https://github.com/googleapis/google-cloud-go/blob/storage/v1.43.0/storage/storage.go#L1338

It's not always set either (needs to be set on upload I think). The reason for the md5 check is so that we can skip files that have already been downloaded correctly.

File: cmd/bootstrap/utils/md5.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package utils
+
+// #nosec
```
Comment (vishalchangrani) [inline cmd/bootstrap/utils/md5.go:3]: thanks for making the linter happy

Comment (vishalchangrani) [inline cmd/bootstrap/utils/md5.go:3]:
```suggestion
// The google storage API only provides md5 and crc32 hence overriding the linter flag for md5
// #nosec
```

**Review by sjonpaulbrown — APPROVED** [review]

**Pull request merged**